src/main/java/com/authservice/repository/UserRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.authservice.repository;
+import com.authservice.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import java.util.Optional;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Integer> {
+    Optional<User> findByName(String username);
```

Comment:
Wrong implemantation -> method name should be findByEmail(String email); Changed in service and cotroller for method name

Reply:
updated code changes : changed findByName(String username) to findByEmail(String email)